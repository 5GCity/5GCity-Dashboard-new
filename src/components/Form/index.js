/**
 * Form Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gptriarca@ubiwhere.com>
 */

import styled from 'styled-components'
import { Form as form } from 'element-react'



const Form = styled(form)`
.el-form-item {
  margin-bottom: 30px;
}
.el-form-item.is-required .el-form-item__label:after {
  content: '*';
  margin-left: 5px;
}
.el-form-item.is-required .el-form-item__label:before {
  content: '';
}
.el-input-group__append, .el-input-group__prepend {
  background: transparent;
  color: ${({theme}) => theme.secondaryColor };
}
.el-input-group__append, .el-input-group__prepend {
  border: none;
}
`
const Item = styled(form.Item)`
.el-form-item__label {
  color: ${({theme}) => theme.secondaryColor };
  font-family: "Open Sans";
  font-size: 14px;
  font-weight: 600;
  line-height: 14px;
}
${({status}) => status && `
  input {
    border: 1px solid rgba(221,108,108,0.4);
  }
  .el-form-item__label {
    color: #DD6C6C;
  }
  .el-form-item.is-required .el-form-item__label:after {
    color: #DD6C6C;
  }
`}
/* ${({status}) => status === 'warning' && `
  input {
    border: 1px solid rgba(221,209,108,0.4);
  }
`}
${({status}) => status === 'success' && `
  input {
    border: 1px solid rgba(140,193,78,0.4);
  }
`} */
${({status}) => !status && `
.el-form-item__label {
  color: ${({theme}) => theme.secondaryColor };
}
`}
`
const Error = styled.div`
  color: ${({theme}) => theme.dangerColor };
  font-size: 12px;
  line-height: 1;
  padding-top: 8px;
  position: absolute;
  top: 100%;
  left: 5px;
`


Form.Item = Item
Form.Error = Error

export default Form
