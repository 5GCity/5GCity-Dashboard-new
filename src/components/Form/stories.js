/**
 * Form Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'

/* Components */
import Input from 'components/Input'
import Form from './index'

const exampleOne = () => (
  <Form labelPosition={'top'} labelWidth='100'>
    <Form.Item label='Name' required status={'warning'}>
      <Input value={''} />
    </Form.Item>
    <Form.Item label='Activity zone' required status={'danger'}>
      <Input
        placeholder='Placeholder Input'
        append={"MB's"}
      />
      <Form.Error>Oops! Error message.</Form.Error>
    </Form.Item>
    <Form.Item label='Activity form' required >
      <Input disabled value={'Another Test'} />
    </Form.Item>
  </Form>
)

storiesOf('Form', module)
  .add('Default', exampleOne)
