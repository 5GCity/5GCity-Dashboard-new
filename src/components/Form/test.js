/**
 * Form Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Form from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Form />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
