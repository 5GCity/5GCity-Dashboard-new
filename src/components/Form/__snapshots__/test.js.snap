// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 .el-form-item {
  margin-bottom: 30px;
}

.c0 .el-form-item.is-required .el-form-item__label:after {
  content: '*';
  margin-left: 5px;
}

.c0 .el-form-item.is-required .el-form-item__label:before {
  content: '';
}

.c0 .el-input-group__append,
.c0 .el-input-group__prepend {
  background: transparent;
}

.c0 .el-input-group__append,
.c0 .el-input-group__prepend {
  border: none;
}

<form
  className="el-form el-form--label-right c0"
  style={Object {}}
/>
`;
